Ginger/Ginger/SolutionWindows/AddApplicationPage.xaml.cs

```diff
@@ -103,18 +111,101 @@ private void OKButton_Click(object sender, RoutedEventArgs e)
             {
                 mSolution.ApplicationPlatforms = new ObservableList<ApplicationPlatform>();
             }
+            var name = variableName.Text.Trim();
+            var description = variableDescription.Text.Trim();
+            if (string.IsNullOrEmpty(name))
+            {
+                NameError.Visibility = Visibility.Visible;
+                return;
+            }
+
+            eUserMsgSelection msgSelection = eUserMsgSelection.None;
+
+            if (!this.isSolutionNew)
+            {
+                msgSelection = Reporter.ToUser(eUserMsgKey.ShareApplicationToEnvironment);
+            }
+
+
+
             foreach (ApplicationPlatform selectedApp in SelectApplicationGrid.Grid.SelectedItems)
             {
                 mSolution.SetUniqueApplicationName(selectedApp);
+
+                selectedApp.AppName = name;
+                selectedApp.Description = description;
+
+
+                if (!this.isSolutionNew && msgSelection.Equals(eUserMsgSelection.Yes))
+                {
+                    var ProjEnvironments = WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<ProjEnvironment>();
+
+                    ProjEnvironments.ForEach((projEnv) => { projEnv.StartDirtyTracking(); projEnv.Applications.Add(new EnvApplication() { Name = selectedApp.AppName, ParentGuid = selectedApp.Guid , Platform = selectedApp.Platform}); });
+
+                }
+
                 mSolution.ApplicationPlatforms.Add(selectedApp);
+
+                var DoesPlatformExist = WorkSpace.Instance.SolutionRepository?.GetAllRepositoryItems<Agent>().Any((agent)=>agent.Platform.Equals(selectedApp.Platform)) ?? true;
```

Comment:
change name DoesPlatformExist to IsMatchingPlatformAgentExist

Reply:
Implemented this comment